unit_test/aestest.cpp

```diff
@@ -194,7 +194,7 @@ void AesTest::OFB256String()
                         "National Institute of Standards and Technology (NIST) in 2001");
     QString key("123456789123");
 
-    QByteArray hashKey = QCryptographicHash::hash(key.toLocal8Bit(), QCryptographicHash::Sha256);
+    QByteArray hashKey = QCryptographicHash::hash(key.toLocal8Bit(), QCryptographicHash::Blake2s_128);
```

Comment:
Are you using Qt6?

Reply:
Is there a reason why you want to replace this test with a 128b one?
If you really want a 128b test, add a test with the correct name and you can use Md5 instead of `Blake2s_128` to generate a 128b hash from the key

Reply:
This actually has a reason.

This test uses AES_128, so the key size should also be 128 bits. I added code in the encrypt and decrypt methods that ensures this:

`QByteArray QAESEncryption::encode(...){
if ((m_mode >= CBC && (iv.isEmpty() || iv.size() != m_blocklen)) || **key.size() != m_keyLen**)
           return QByteArray();`

so: using SHA-256 here results in the encryption not working.
and I used Blake2s_128 for no other reason than I didn't think of Md5.